modules/llrt_tty/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use llrt_utils::module::{export_default, ModuleInfo};
+use rquickjs::{
+    module::{Declarations, Exports, ModuleDef},
+    prelude::Func,
+    Ctx, Result,
+};
+
+fn isatty(fd: i32) -> bool {
+    unsafe { libc::isatty(fd) != 0 }
+}
+
+pub struct TtyModule;
+
+impl ModuleDef for TtyModule {
+    fn declare(declare: &Declarations<'_>) -> Result<()> {
+        declare.declare("isatty")?;
+        declare.declare("default")?;
+        Ok(())
+    }
+
+    fn evaluate<'js>(ctx: &Ctx<'js>, exports: &Exports<'js>) -> Result<()> {
+        export_default(ctx, exports, |default| {
+            default.set("isatty", Func::from(isatty))?;
+            Ok(())
+        })
+    }
+}
+
+impl From<TtyModule> for ModuleInfo<TtyModule> {
+    fn from(val: TtyModule) -> Self {
+        ModuleInfo {
+            name: "tty",
+            module: val,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::TtyModule;
+    use llrt_test::{call_test, test_async_with, ModuleEvaluator};
+
+    #[tokio::test]
+    async fn test_isatty() {
+        test_async_with(|ctx| {
+            Box::pin(async move {
+                ModuleEvaluator::eval_rust::<TtyModule>(ctx.clone(), "tty")
+                    .await
+                    .unwrap();
+
+                let module = ModuleEvaluator::eval_js(
+                    ctx.clone(),
+                    "test",
+                    r#"
+                        import { isatty } from 'tty';
+
+                        export async function test() {
+                            return new Array(5).fill(0).map((_, i) => +isatty(i)).join('')
+                        }
+                    "#,
+                )
+                .await
+                .unwrap();
+                let expect = if std::env::var("GITHUB_ACTIONS").unwrap_or_default() == "true" {
```

Comment:
We can't hard code GITHUB_ACTIONS to test this. Better would be to run the `test -t` command as a subcommand and test this on posix only for now. See comment

Reply:
I'm not sure if the two are equivalent, but using is_terminal removes libc

```
fn isatty(fd: i32) -> bool {
    unsafe { libc::isatty(fd) != 0 }
}
```

```
fn isatty(fd: i32) -> bool {
    match fd {
        0 => stdin().is_terminal(),
        1 => stdout().is_terminal(),
        2 => stderr().is_terminal(),
        _ => false,
    }
}
```